src/hugr/region/petgraph.rs

```diff
@@ -12,22 +13,34 @@ use portgraph::NodeIndex;
 
 macro_rules! impl_region_petgraph_traits {
     ($hugr:ident) => {
-        impl<'a> petgraph::visit::GraphBase for $hugr<'a> {
+        impl<'a, Base> petgraph::visit::GraphBase for $hugr<'a, Base>
```

Comment:
nit: I wonder about something like `use petgraph::visit as pv` for all of these (and maybe get all on one line with the constraints)